src/map.c

```diff
@@ -2180,9 +2179,21 @@ put_escstr(FILE *fd, char_u *strstart, int what)
 	    {
 		modifiers = str[2];
 		str += 3;
-		c = *str;
+
+		// Modifiers can be applied too to multi-byte characters.
+		p = mb_unescape(&str);
+
+		if (p == NULL)
+		{
+		    // manually advance str if mb_unescape() fails
+		    p = str;
+		    str += (*mb_ptr2len)(str);
+		}
+
+		c = (*mb_ptr2char)(p);
+		--str;
 	    }
-	    if (c == K_SPECIAL)
+	    else if (c == K_SPECIAL)
```

Comment:
Adding this `else` doesn't seem correct. A modifier can be followed by a special key (e.g. `<C-Right>`).

Reply:
I think `else` is justified because, if the first conditional branch is executed, `c` is now *unescaped* and cannot be a special character (is a *code point* or `cell` like vim docs state).

Reply:
No, if `mb_unescape()` fails, you need to check if `*str == K_SPECIAL`.

Reply:
You are right. `mb_unescape()` will fail for ascii characters and `K_SPECIAL` or `CSI` (if not a *escaping* case).
The `else` must be removed.